src/jdk.unsupported/share/classes/sun/misc/Unsafe.java

```diff
@@ -651,7 +651,7 @@ public long objectFieldOffset(Field f) {
         if (declaringClass.isHidden()) {
             throw new UnsupportedOperationException("can't get field offset on a hidden class: " + f);
         }
-        if (f.getDeclaringClass().isPrimitiveClass()) {
+        if (PrimitiveClass.isPrimitiveClass(f.getDeclaringClass())) {
             throw new UnsupportedOperationException("can't get field offset on an inline class: " + f);
```

Comment:
since you are in this file, can you update the error message `s/inline class/primitive class/`